pkg/cmd/server/server.go

```diff
@@ -228,6 +229,7 @@ type server struct {
 	dynamicClient                       dynamic.Interface
 	sharedInformerFactory               informers.SharedInformerFactory
 	csiSnapshotterSharedInformerFactory *CSIInformerFactoryWrapper
+	csiSnapshotClient                   *snapshotterClientSet.Clientset
```

Comment:
Why use the client here instead of using the informer factory & lister? It seems like we're introducing 2 ways of doing the same thing in different controllers.

Reply:
In the deletion controller, in addition to listing volumesnapshots  we also want to patch volumesnapshotcontents, and delete volumesnapshots. I wasn't able to use the informer factory APIs to do that or to get it to expose the underlying client.
That's the reason I chose to embed the client in the server struct.

However, we can embed the client in the server struct and use that to create the informer factory in the backup controller. I was going to make that refactor in a different PR.

Reply:
Got it, thanks. Could you please add a comment explaining that, just so it's more immediately findable by reading the code?

I don't know if we want to create the informer factory in the controller. There's some warnings about that in the server.go file about doing that kind of thing, but it's possible that could be an optimization.

Reply:
Yeah will add comments.

Reply:
Nolan's right that it would be more efficient to use the lister for the query inside the deletion controller. However, you will also need the client in order to do the patching & deletion. So it's not an either/or, but a "both".

You wouldn't want to create a new informer factory inside the controller - use the existing factory in `server.go` to get a VS/VSC lister, and pass those into the constructor for the controller, along with the client.

Reply:
Makes sense.
But this is so hard to write unit tests for!
```
# github.com/vmware-tanzu/velero/pkg/controller
import cycle not allowed in test
package github.com/vmware-tanzu/velero/pkg/controller (test)
	imports github.com/vmware-tanzu/velero/pkg/cmd/server
	imports github.com/vmware-tanzu/velero/pkg/controller
```